src/sce-elf.c

```diff
@@ -695,8 +786,14 @@ void *sce_elf_module_info_encode(
 	module_info_raw->type = module_info->type;
 	module_info_raw->export_top = htole32(OFFSET(sceLib_ent));
 	module_info_raw->export_end = htole32(OFFSET(sceLib_ent) + sizes->sceLib_ent);
+	sce_elf_export_sort(module_info);
+
 	module_info_raw->import_top = htole32(OFFSET(sceLib_stubs));
 	module_info_raw->import_end = htole32(OFFSET(sceLib_stubs) + sizes->sceLib_stubs);
+	// if want sce_elf_import_code_address_sanitization, need working in very low level
+	// sce_elf_import_code_address_sanitization(module_info);
```

Comment:
explain this better

Reply:
The imported function is already assigned an address, so sanitizing the import address here will result in an unexpected call in the vita program.

Reply:
what's the sanitizing process?

Reply:
Simple example

```
//SceThreadmgrForDriver:0xe2c40624
//0: ksceKernelSetPermission         0x2eedf17 @OFF: 0xb0a0 VADDR: 0xb0a0
//1: ksceKernelSetProcessId          0x486f239 @OFF: 0xb0b0 VADDR: 0xb0b0
//2: ksceKernelExitDeleteThread     0x1d17decf @OFF: 0xb050 VADDR: 0xb050
//3: SceThreadmgrForDriver_20C228E4 0x20c228e4 @OFF: 0xb0d0 VADDR: 0xb0d0
//4: SceThreadmgrForDriver_3A72C6D8 0x3a72c6d8 @OFF: 0xb0e0 VADDR: 0xb0e0
//5: ksceKernelGetSystemTimeLow     0x47f6de49 @OFF: 0xb070 VADDR: 0xb070
//6: ksceKernelDelayThread          0x4b675d05 @OFF: 0xb040 VADDR: 0xb040
//7: ksceKernelGetThreadId          0x59d06540 @OFF: 0xb090 VADDR: 0xb090
//8: ksceKernelGetProcessId         0x9dcb4b7a @OFF: 0xb060 VADDR: 0xb060
//9: ksceKernelEnqueueWorkQueue     0xe50e1185 @OFF: 0xb0c0 VADDR: 0xb0c0
//10: ksceKernelGetSystemTimeWide   0xf4ee4fa9 @OFF: 0xb080 VADDR: 0xb080
```

to

```
//SceThreadmgrForDriver:0xe2c40624
//0: ksceKernelSetPermission         0x2eedf17 @OFF: 0xb040 VADDR: 0xb040
//1: ksceKernelSetProcessId          0x486f239 @OFF: 0xb050 VADDR: 0xb050
//2: ksceKernelExitDeleteThread     0x1d17decf @OFF: 0xb060 VADDR: 0xb060
//3: SceThreadmgrForDriver_20C228E4 0x20c228e4 @OFF: 0xb070 VADDR: 0xb070
//4: SceThreadmgrForDriver_3A72C6D8 0x3a72c6d8 @OFF: 0xb080 VADDR: 0xb080
//5: ksceKernelGetSystemTimeLow     0x47f6de49 @OFF: 0xb090 VADDR: 0xb090
//6: ksceKernelDelayThread          0x4b675d05 @OFF: 0xb0a0 VADDR: 0xb0a0
//7: ksceKernelGetThreadId          0x59d06540 @OFF: 0xb0b0 VADDR: 0xb0b0
//8: ksceKernelGetProcessId         0x9dcb4b7a @OFF: 0xb0c0 VADDR: 0xb0c0
//9: ksceKernelEnqueueWorkQueue     0xe50e1185 @OFF: 0xb0d0 VADDR: 0xb0d0
//10: ksceKernelGetSystemTimeWide   0xf4ee4fa9 @OFF: 0xb0e0 VADDR: 0xb0e0
```

Reply:
> @frangarcj @yne could you review this additionally?

I'd prefer keeping the codebase simple over mimicking every aspect of the official SDK.

But maybe I'm missing some sheme plan that would justify this mimicking (like rebuilding kernel modules from RE sources ?).

Reply:
For example, in the nid lookup system, a better algorithm can be implemented if the nids are sorted neatly.